### PR TITLE
feat: Add "What is Machine Learning" page to the handbook

### PR DESCRIPTION
**Pull request by ShagReza**

### CODE REVIEW

File: docs/disciplines/Machine_Learning/index.md [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+- When data is insufficient or of poor quality.
+- When simpler, rule-based systems can solve the problem more effectively.
+- When interpretability is critical, and black-box methods aren't acceptable.
+
+## Categories and Branches of Machine Learning
+
+ML is broadly divided into three main categories:
+
+1. **Supervised Learning**:
+   - Models are trained on labeled data (input-output pairs).
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:48]: Fix the nested list formatting, the indented points need an extra space (4 spaces total)
```suggestion

    - Models are trained on labeled data (input-output pairs).
```

File: docs/disciplines/Machine_Learning/index.md [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
+   - Examples: Linear regression, support vector machines (SVMs).
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:49]:
```suggestion
    - Examples: Linear regression, support vector machines (SVMs).
```

File: docs/disciplines/Machine_Learning/index.md [hunk 3]
```diff
@@ -1,4 +1,58 @@
...
+
+2. **Unsupervised Learning**:
+   - Models learn patterns from unlabeled data.
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:52]:
```suggestion

    - Models learn patterns from unlabeled data.
```

File: docs/disciplines/Machine_Learning/index.md [hunk 4]
```diff
@@ -1,4 +1,58 @@
...
+   - Examples: Clustering (e.g., k-means), dimensionality reduction (e.g., PCA).
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:53]:
```suggestion
    - Examples: Clustering (e.g., k-means), dimensionality reduction (e.g., PCA).
```

File: docs/disciplines/Machine_Learning/index.md [hunk 5]
```diff
@@ -1,4 +1,58 @@
...
+
+3. **Reinforcement Learning**:
+   - Models learn by interacting with an environment and receiving rewards or penalties.
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:56]:
```suggestion

    - Models learn by interacting with an environment and receiving rewards or penalties.
```

File: docs/disciplines/Machine_Learning/index.md [hunk 6]
```diff
@@ -1,4 +1,58 @@
...
+   - Examples: Markov decision process (MDP), Deep Q Networks (DQN).
```
Comment (mtran-code) [inline docs/disciplines/Machine_Learning/index.md:57]:
```suggestion
    - Examples: Markov decision process (MDP), Deep Q Networks (DQN).
```

**Review by mtran-code — CHANGES_REQUESTED** [review]
This was really detailed and informative, thanks!

File: docs/disciplines/Machine_Learning/index.md (hunk 6 above)
Comment (nabinelnino) [inline docs/disciplines/Machine_Learning/index.md:57]:
```suggestion
## Categories and Branches of Machine Learning

### Supervised Learning:

- Models are trained on labeled data (input-output pairs).
- Examples: Linear regression, support vector machines (SVMs).

### Unsupervised Learning:

- Models learn patterns from unlabeled data.
- Examples: Clustering (e.g., k-means), dimensionality reduction (e.g., PCA).

### Reinforcement Learning:

- Models learn by interacting with an environment and receiving rewards or penalties.
- Examples: Markov decision process (MDP), Deep Q Networks (DQN).
```suggestion
## Categories and Branches of Machine Learning

ML is broadly divided into three main categories:

1. **Supervised Learning**:
   - Models are trained on labeled data (input-output pairs).
   - Examples: Linear regression, support vector machines (SVMs).

2. **Unsupervised Learning**:
   - Models learn patterns from unlabeled data.
   - Examples: Clustering (e.g., k-means), dimensionality reduction (e.g., PCA).

3. **Reinforcement Learning**:
   - Models learn by interacting with an environment and receiving rewards or penalties.
   - Examples: Markov decision process (MDP), Deep Q Networks (DQN).
```

**Review by mtran-code — APPROVED** [review]

**Pull request merged**